src/mam4xx/aero_model_emissions.hpp

```diff
@@ -910,62 +947,79 @@ void aero_model_emissions(
 
-  init_dust_dmt_vwr(dust_data.dust_dmt_grd, dust_data.dust_dmt_vwr);
   dust_emis(
       // in
-      dust_indices, dust_density, dust_flux_in, dust_data, soil_erodibility,
+      dust_density, dust_flux_in, dust_data, soil_erodibility,
       // inout
       cflux);
+
   init_seasalt(seasalt_data);
+
   calculate_seasalt_numflux_in_bins(
       // in
       surface_temp, u_bottom, v_bottom, z_bottom, seasalt_data.consta,
       seasalt_data.constb,
       // out
       fi);
+
   seasalt_emis(
       // in
       fi, ocean_frac, seasalt_emis_scalefactor, seasalt_data,
       //  inout
       cflux);
+
+  // BAD CONSTANTS
+  constexpr bool emit_this_mode[om_num_modes] = {true, true, false};
+
   marine_organic_emissions(
       // in
       fi, ocean_frac, seasalt_emis_scalefactor, seasalt_data, emit_this_mode,
       // inout
       cflux);
-} // end aero_model_emissions()
 
-KOKKOS_INLINE_FUNCTION
-void aero_model_emissions(
-    // in
-    OnlineEmissionsData online_emiss_data, SeasaltEmissionsData seasalt_data,
-    DustEmissionsData dust_data,
-    // inout
-    // NOTE: fortran: cam_in%cflx
-    View1D &cflux_) {
-
-  Real cflux[pcnst];
-  for (int i = 0; i < pcnst; ++i) {
-    cflux[i] = cflux_(i);
-  }
-  aero_model_emissions(online_emiss_data, seasalt_data, dust_data, cflux);
 } // end aero_model_emissions()
 
 KOKKOS_INLINE_FUNCTION
-void aero_model_emissions(
-    // inout
-    View1D &cflux_) {
-
+void aero_model_emissions(const Real &sst, const Real &ocnfrac,
+                          const Real &u_bottom, const Real &v_bottom,
+                          const Real &z_bottom, const const_view_1d &dstflx,
+                          const Real &soil_erodibility, const Real &mpoly,
+                          const Real &mprot, const Real &mlip,
+                          // inout
+                          view_1d &cflux_) {
+
+  // Populate online emission data structure with inputs
   OnlineEmissionsData online_emiss_data;
+  for (int idst = 0; idst < dust_nflux_in; ++idst) {
+    online_emiss_data.dust_flux_in[idst] = dstflx(idst);
+  }
+  online_emiss_data.surface_temp = sst;
+  online_emiss_data.u_bottom = u_bottom;
+  online_emiss_data.v_bottom = v_bottom;
+  online_emiss_data.ocean_frac = ocnfrac;
+  online_emiss_data.z_bottom = z_bottom;
+  online_emiss_data.soil_erodibility = soil_erodibility;
+
+  // Populate Seasalt emissions data structure with inputs
   SeasaltEmissionsData seasalt_data;
+  seasalt_data.mpoly = mpoly;
+  seasalt_data.mprot = mprot;
+  seasalt_data.mlip = mlip;
+
   DustEmissionsData dust_data;
+
+  // initialize dust_dmt_vwr data
+  init_dust_dmt_vwr(dust_data);
+
+  // FIXME: why not send it as an input as a view????
   Real cflux[pcnst];
   for (int i = 0; i < pcnst; ++i) {
     cflux[i] = cflux_(i);
   }
-  aero_model_emissions(online_emiss_data, seasalt_data, dust_data, cflux);
+  aero_model_emissions(online_emiss_data, seasalt_data, dust_data,
+                       cflux); // out
```

Comment:
@singhbalwinder : Do you need to copy the local value of cflux to the cflux_ Kokkos view after the call to aero_model_meeisions?